server/logger.js

```diff
@@ -0,0 +1,57 @@
+/*
+ ** When importing the logger the module name should be add to the logger
+ ** Example import:
+ ** const logger = require('./logger')('server.js');
+ ** 'server.js' been the file where the logger is been imported into.
+ */
+
+const { createLogger, format, transports } = require('winston');
+
+const messageFormat = format.printf(
+  ({ level, label, message, timestamp }) => `${timestamp}: ${label} - ${level}: ${message}`
+);
+
+module.exports = function(fileName) {
+  const defaultConsoleFormat = format.combine(
+    format.colorize({ all: true }),
+    format.timestamp(),
+    format.label({ label: fileName }),
+    messageFormat
+  );
+
+  const options = {
+    info: {
+      console: {
+        level: 'info',
+        handleExceptions: true,
+        format: defaultConsoleFormat
+      }
+    },
+    debug: {
+      console: {
+        level: 'debug',
+        handleExceptions: true,
+        format: defaultConsoleFormat
+      },
+      file: {
+        filename: 'dev-debug.log',
+        level: 'debug',
+        handleExceptions: true,
+        format: format.combine(format.timestamp(), format.label({ label: fileName }), messageFormat)
+      }
+    }
+  };
+
+  const logger = createLogger();
+
+  if (process.env.NODE_ENV !== 'production') {
```

Comment:
We will log to the console most of the time, so I would configure the `file transport` as optional
```suggestion
 if (process.env.NODE_ENV === 'development') {
    logger.add(new transports.Console(options.debug.console));
    logger.add(new transports.File(options.debug.file)); // creates local dev-debug.log file
  } else {
    logger.add(new transports.Console(options.info.console));
  }
```

Why do we have a `options.debug.console` and a `options.info.console` ? They looks identical

Reply:
Ok I will change this. I just want to point out that the `logger.verbose` calls will only show in the `debug` config. If required at a later stage we could add an extra flag for this.

Reply:
I missed your question about the `options.debug.console` and options.info.console`. The difference is the log level which is shown in the console. `options.debug.console` will log all levels from `debug` up while `options.info.console` only logs from `info` up which means `debug` and `verbose` are not logged to the console.